Patch-Source: https://gitlab.gnome.org/World/Chatty/-/merge_requests/1473
---
From 52690dc74dd6f09a5dbdcd2424d63a65661d930d Mon Sep 17 00:00:00 2001
From: fossdd <fossdd@pwned.life>
Date: Fri, 25 Apr 2025 16:51:49 +0200
Subject: [PATCH] data: Use APP_ID for metainfo ID

Fixes the mismatch of app id, desktop/dbus service file.

Rather than having multiple app ids (lowercase and not) replace the
metainfo ID with the APP_ID to fix the appstreamcli validate-tree
warning and keep the application ID everywhere consistent.

This reverts commit 70d1ba4ebaa43937dd304b733ceed9c8c0296900.
---
 data/meson.build                       | 7 +++----
 data/sm.puri.Chatty.metainfo.xml.in.in | 4 ++--
 2 files changed, 5 insertions(+), 6 deletions(-)

diff --git a/data/meson.build b/data/meson.build
index 1e75215e..0206dd54 100644
--- a/data/meson.build
+++ b/data/meson.build
@@ -1,6 +1,5 @@
 config_data = configuration_data()
 config_data.set('APP_ID', app_id)
-config_data.set('LOWERCASE_APP_ID', app_id.to_lower())
 
 desktop_file = i18n.merge_file(
   input: configure_file(
@@ -8,7 +7,7 @@ desktop_file = i18n.merge_file(
     output: '@0@.desktop.in'.format(app_id),
     configuration: config_data,
   ),
-  output: '@0@.desktop'.format(app_id.to_lower()),
+  output: '@0@.desktop'.format(app_id),
   type: 'desktop',
   po_dir: '../po',
   install: true,
@@ -21,7 +20,7 @@ desktop_file = i18n.merge_file(
     output: '@0@-daemon.desktop.in'.format(app_id),
     configuration: config_data,
   ),
-  output: '@0@-daemon.desktop'.format(app_id.to_lower()),
+  output: '@0@-daemon.desktop'.format(app_id),
   type: 'desktop',
   po_dir: '../po',
   install: true,
@@ -41,7 +40,7 @@ metainfo_file = i18n.merge_file(
     output: '@0@.metainfo.xml.in'.format(app_id),
     configuration: config_data,
   ),
-  output: '@0@.metainfo.xml'.format(app_id.to_lower()),
+  output: '@0@.metainfo.xml'.format(app_id),
   po_dir: '../po',
   install: true,
   install_dir: join_paths(get_option('datadir'), 'metainfo')
diff --git a/data/sm.puri.Chatty.metainfo.xml.in.in b/data/sm.puri.Chatty.metainfo.xml.in.in
index 5eaf93bf..d176e934 100644
--- a/data/sm.puri.Chatty.metainfo.xml.in.in
+++ b/data/sm.puri.Chatty.metainfo.xml.in.in
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <component type="desktop-application">
-    <id>@LOWERCASE_APP_ID@</id>
+    <id>@APP_ID@</id>
     <metadata_license>CC0-1.0</metadata_license>
     <project_license>GPL-3.0-or-later</project_license>
     <name>Chats</name>
@@ -14,7 +14,7 @@
     <url type="translate">https://l10n.gnome.org/module/chatty/</url>
     <translation type="gettext">purism-chatty</translation>
     <update_contact>librem5-dev@lists.community.puri.sm</update_contact>
-    <launchable type="desktop-id">@LOWERCASE_APP_ID@.desktop</launchable>
+    <launchable type="desktop-id">@APP_ID@.desktop</launchable>
     <developer id="chatty.puri.sm">
       <name translate="no">The Chatty Team</name>
     </developer>
-- 
GitLab

